### ISSUE TITLE
Ok to run dada2 on samples generated with multiple primers (like Swift 16S/ITS kits)?

### ISSUE THREAD

**Issue by AstrobioMike** [benjjneb/dada2#1370]
Hey there :) 

I've come across a [Swift 16S+ITS kit](https://swiftbiosci.com/swift-amplicon-16s-its-panel/)-produced dataset that I need to deal with for the first time. All I've found online so far is this [workflow on github](https://github.com/swiftbiosciences/16S-SNAPP-py3) from the company, that seems to run samples through dada2 (after primer-trimming) despite each sample being comprised of mixtures of different regions of 16S. 

I had initially thought this wouldn't work because I assumed all the reads being given to dada2 for a given sample should be from the same region/primers, so it is learning errors and inferring ASVs across all of them by looking at the same target region. But then I started confusing myself and wondering if maybe it doesn't actually matter so long as there are many of each unique target region...

**Is it okay to run things like `learnErrors` and the `dada` inference step on a sample that contains a mixture of different regions and even possibly different target genes?**

**Comment by nagasridhar** [comment]
Hi Mike,

I have been exploring the Swift 16S+ITS kit as well. From my interactions
with the company, they said that their workflow on github uses the primers
to filter out either the 16S or ITS regions based on the workflow you
choose, i.e., q2_ITS and 16S_SNAPP.

I have tested the data from Swift kit using an alternative approach with
good results,

https://astrobiomike.github.io/amplicon/16S_and_18S_mixed. I replaced the
18S database with the ITS database from NCBI to separate out the reads into
16S and ITS regions and then do the analysis.

In addition, I have also seen another way of dealing with the data after
separating the 16S and ITS regions, especially 16S regions:
https://q2-sidle.readthedocs.io/en/latest/.

In short, I think separating the data into their target regions and then
doing the analysis would be better since the error plots for the regions
differ and this may influence your final ASV table.

Ben may provide more input regarding how to deal with these datasets. I
look forward to hearing Ben's thoughts as well.

**Comment by AstrobioMike** [comment]
Thanks for your thoughts and experience, @nagasridhar :) 

>  From my interactions with the company, they said that their workflow on github uses the primers to filter out either the 16S or ITS regions based on the workflow you choose, i.e., q2_ITS and 16S_SNAPP.

Does that mean for processing their 16S+ITS kit, the company suggested to you to just run the mixed dataset through 1 of those workflows and then also run the full dataset again through the other? (with the ITS one being in qiime and the 16S one not)

**Comment by nagasridhar** [comment]
Oops, I did not realize I was replying to you @AstrobioMike. Its funny I recommended your pipeline to you! My apologies!


> Does that mean for processing their 16S+ITS kit, the company suggested to you to just run the mixed dataset through 1 of those workflows and then also run the full dataset again through the other? (with the ITS one being in qiime and the 16S one not)

Yes, that is what they recommended to me.

**Comment by AstrobioMike** [comment]
> Oops, I did not realize I was replying to you @AstrobioMike. Its funny I recommended your pipeline to you! My apologies!

Haha, fortunately it wasn't in the vein of "definitely don't do this!". It made my day :) 

Thanks again for taking the time to share your experience, @nagasridhar!

Like you noted, I also hope to hear @benjjneb's thoughts as well. Really just on this core concept now: 

**Is it okay to run things like `learnErrors` and the `dada` inference step on a sample that contains a mixture of different target regions?**

**Comment by benjjneb** [comment]
> Is it okay to run things like learnErrors and the dada inference step on a sample that contains a mixture of different target regions?

In principle, yes. For DADA2, it's not different than if the two different target regions were just two different disjoint sets of unique ASVs from the same target region.

In practice, it's important to be sure that amplicons from all target regions don't have primers/adapters still attached, and that the PCR step is reasonably consistent between them as that is a non-negligible source of errors.

**Comment by AstrobioMike** [comment]
Thanks for the confirmation, Ben! And thanks again, @nagasridhar!
